src/components/InfiniteRelayList.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-unused-vars */
+import { isEqual } from "lodash";
+import React from "react";
+
+import { emptyRelayPage, RelayPaginatedResponse } from "../api/pagination";
+import apolloClient from "../network/apollo-client-singleton";
+import Fragment from "./Fragment";
+import WhenSeen from "./WhenSeen";
+
+interface InfiniteRelayListProps<T> {
+  query: any;
+  queryVars: Record<string, any>;
+  nextQueryVars(cursor: string | null): Record<string, any>;
+  renderNode: (node: T, idx: number) => React.ReactElement;
+  toRelay: (data: any) => RelayPaginatedResponse<T>;
+  keyFunc?: (node: T, idx: number) => any;
+  cliffHanger?: (last: RelayPaginatedResponse<T>) => React.ReactElement;
+  empty?: () => React.ReactElement;
+  dbLastUpdatedAt?: Date;
+}
+
+interface InfiniteRelayListState<T> {
+  nodes: T[];
+  last: RelayPaginatedResponse<T>;
+  loading: boolean;
+  hasMore: boolean;
+}
+
+class InfiniteRelayList<T> extends React.Component<
+  InfiniteRelayListProps<T>,
+  InfiniteRelayListState<T>
+> {
+  state: InfiniteRelayListState<T> = {
+    nodes: [],
+    last: emptyRelayPage<T>(),
+    loading: false,
+    hasMore: false
+  };
+
+  async componentDidMount() {
+    await this.load();
+  }
+
+  componentDidUpdate(prevProps: InfiniteRelayListProps<T>) {
+    if (
+      prevProps.query !== this.props.query ||
+      !isEqual(prevProps.queryVars, this.props.queryVars) ||
+      !isEqual(prevProps.dbLastUpdatedAt, this.props.dbLastUpdatedAt)
+    ) {
+      this.load();
+    }
+  }
+
+  async query(
+    variables = this.props.queryVars
+  ): Promise<RelayPaginatedResponse<T>> {
+    const { data } = await apolloClient.query({
+      query: this.props.query,
+      variables,
+      fetchPolicy: "no-cache"
```

Comment:
do you have a preference as to how? in the case of the people admin page, after editing a single person, the cache should be invalidated for the "current page x filter+query", but ideally, the last cache result would be used if `dbLastUpdatedAt ` hasn't changed.  keeping track of queries x pages  x what their `dbLastUpdatedAt` value was when last cache value was pulled didn't seem to me like it'd add much of an advantage in the current use case. it also seemed to me like if the apollo-client version was ever upgraded, there'd be `cache-and-network` which would be a low-code-addition way to get use-the-cache-for-fast-ui-when-possible anyways.

Reply:
i'd also add it took me a day of debugging to finally figure out the reason i was only sometimes seeing new person updates was because of this, i.e. the apollo cache was working as intended. so it seemed better to just make this component "everything will take 20-100 ms more" by default, but messy to make it externally configurable (i think if i was putting in an infinite scroll list, saw the name of this component, then saw some kind of cache-related prop, i'd probably just say yes to any caching and then forget about it again).

Reply:
I think a default no-cache here is fine

Reply:
okay i'm down with that. just tried `network-only` again (i had originally gone with it instead of no-cache but it looked like it wasn't updating the ui. i think because i had my onEdit callbacks wrong though).